### PR TITLE
fix: [#186429361] dynamics license status application id filtering to…

### PR DESCRIPTION
**Pull request by somabadri**
… account for null

## Description
Updated filtering to account for potential null values for rgb_number and statecode. 

### Ticket

[#186429361](https://www.pivotaltracker.com/story/show/186429361)

## Code author checklist

- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

**Review by adidner — APPROVED** [review]
Seems good to me

File: api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts [hunk 1]
```diff
@@ -38,10 +38,11 @@ export const DynamicsLicenseApplicationIdClient = (
         const mainActiveApplications = response.data.value
           .filter(
             (applicationDetails: LicenseApplicationIdApiResponse) =>
-              applicationDetails.statecode === ACTIVE_STATECODE
+              applicationDetails.statecode !== null && applicationDetails.statecode === ACTIVE_STATECODE
           )
           .filter(
             (applicationDetails: LicenseApplicationIdApiResponse) =>
+              applicationDetails.rgb_number !== null &&
```
Comment (seidior) [inline api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts:45]: [pebble] Are you able to leverage JS truthiness for both of these tests? Something along the lines of:

```typescript
applicationDetails.rgb_number && ...
```

File: api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts [hunk 2]
```diff
@@ -38,10 +38,11 @@ export const DynamicsLicenseApplicationIdClient = (
         const mainActiveApplications = response.data.value
           .filter(
             (applicationDetails: LicenseApplicationIdApiResponse) =>
-              applicationDetails.statecode === ACTIVE_STATECODE
+              applicationDetails.statecode !== null && applicationDetails.statecode === ACTIVE_STATECODE
```
Comment (seidior) [inline api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts:41]: [pebble] If `applicationDetials.stateCode === ACTIVE_STATECODE`, then it's necessarily not null. What do we gain by adding in a not-null test?

**Review by seidior — APPROVED** [review]
Looking good here, Soma! Just a couple of comments for your review.

File: api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts (hunk 1 above)
Comment (somabadri) [reply api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts]: That's a good point, changed it!

File: api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts (hunk 2 above)
Comment (somabadri) [reply api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts]: I guess it doesn't provide the same value as the other null test as we aren't trying to manipulate the value in anyway, got rid of it.

**Pull request merged**